### PR TITLE
Add docs for GHA self-hosted settings

### PR DESCRIPTION
**Pull request by isuruf**
PR Checklist:

- [ ] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] put any other relevant information below

See https://github.com/conda-forge/conda-smithy/pull/1703

### CODE REVIEW

File: src/maintainer/conda_forge_yml.rst [hunk 1]
```diff
@@ -353,6 +354,50 @@ defaults are as follows:
       # conda-forge-ci-setup-feedstock references
       tooling_branch_name: main
 
+.. _github_actions:
+
+github_actions
```
Comment (jaimergp) [inline src/maintainer/conda_forge_yml.rst:359]: Should we add a note about how MS-hosted GHA workflows are not currently supported for feedstocks, beyond a few specific core ones?

Comment (isuruf) [reply src/maintainer/conda_forge_yml.rst:359]: Done.

**Review by jaimergp — APPROVED** [review]

**Review by beckermr — APPROVED** [review]

**Comment by jaimergp** [comment]
I'm going to merge this for now. There's a chance we need to expand a bit on the details of GPU runners but for now this is valid information even if it's in a general way for any kind of self-hosted GHA runners.

**Comment by hmaarrfk** [comment]
so can i donate a computer of mine to conda-forge to build some packages for us ;) or is this for non-conda-forge users to setup their own self hosted stuff?

File: src/maintainer/conda_forge_yml.rst [hunk 2]
```diff
@@ -353,6 +354,50 @@ defaults are as follows:
...
+   github_actions_labels:
+     # use Microsoft free runners
+     - hosted                                   # [osx or win]
+     # Use self-hosted runner with custom label
+     - - self-hosted                            # [linux and aarch64]
+       - custom-label                           # [linux and aarch64]
+     # Use self-hosted runner from cirun
+     - cirun-openstack-cpu-large                # [linux and ppc64le]
+     # Use self-hosted gpu runner from cirun
+     - cirun-openstack-gpu-large                # [linux and x86_64]
```
Comment (adrianinsaval) [inline src/maintainer/conda_forge_yml.rst:398]: Could somebody explain how do I interpret this? how do I use this? how can I achieve this output:
```
          - CONFIG: osx_arm64_python3.11.____cpython
            UPLOAD_PACKAGES: True
            os: macos
            runs_on: ['macos-14']
```

macos-14 are github's new arm macos runners

Comment (jaimergp) [reply src/maintainer/conda_forge_yml.rst:398]: You can't, this is only for self-hosted runners, or organizations that are not conda-forge. We haven't enabled native osx-arm64 runners with Github. See discussion at #1781

Comment (adrianinsaval) [reply src/maintainer/conda_forge_yml.rst:398]: I'm trying to do it for a not conda-forge feedstock. Would be nice if smithy woudl support that in such cases, loks like most of the building blocks needed for such functionality are there. It's a PITA manually changing back to the specific runner I want every time I rerender. I also want to specify the macos version for the x86_64 runner, my builds hang for a prolonged time sometimes until timeout on macos-latest but usually run without issue on macos-11

Comment (jakirkham) [reply src/maintainer/conda_forge_yml.rst:398]: Should we convert this to [a conda-smithy issue]( https://github.com/conda-forge/conda-smithy )?

Would help if we can capture what exactly is being tried, what happens now, and what the use case needs to work in the ideal case

**Pull request merged**